astroid/__init__.py

```diff
@@ -180,7 +180,7 @@
 
 # isort: on
 
-from astroid.util import Uninferable
+from astroid.util import Uninferable, UninferableBase
```

Comment:
I would probably skip adding this here.

Reply:
This is "necessary" for `pylint` I think, but perhaps also a good question to ask now.
What do we want to do in `pylint` use `node: UninferableBase` or `node: type[Uninferable]`? Some goes for checking `node is Uninferable` or `isinstance(node, UninferableBase)`.

Reply:
Definitely `UninferableBase`. You pointed out above the trouble with `type[Uninferable]`. (The pyright dev said it was straight wrong.)

> Some goes for checking node is Uninferable or isinstance(node, UninferableBase)?

Either. The latter is safer against copying/unpickling.

> This is "necessary" for pylint I think

I think I'm missing this part; do you mind spelling it out? It could be imported from `astroid.util` instead of `astroid`.

Reply:
Ah yeah, but I thought we put all things that are public as classes directly in `astroid`. But I do prefer importing from `util` myself so I'll remove this.

Reply:
> Ah yeah, but I thought we put all things that are public as classes directly in astroid.

I think it's a totally random and difficult to maintain snapshot of shortcuts. I'd like to stop adding to it, if possible:

```python
>>> import astroid
>>> astroid.PY310_PLUS
False
>>> astroid.PY311_PLUS
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'astroid' has no attribute 'PY311_PLUS'
```

Reply:
importing from astroid directly has the advantage that we can refactor everything the way we want without breaking the API. And astroid has some really nasty construct and import cycle internally that makes this valuable.